### PR TITLE
Changed ReindexRequest to use Writeable.Reader

### PR DESCRIPTION
**Pull request by sohaibiftikhar**
-- This is a pre-stage for adding the reindex API to the REST high-level-client
-- Follows the pattern set in #26315

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]
Looks right to me.

@elastisticmachine, whitelist this please

**Comment by nik9000** [comment]
white list this please

**Comment by nik9000** [comment]
add to whitelist

File: server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java [hunk 1]
```diff
@@ -59,6 +59,13 @@ private ReindexRequest(SearchRequest search, IndexRequest destination, boolean s
         this.destination = destination;
     }
 
+    public ReindexRequest(StreamInput in) throws IOException {
+        super.readFrom(in);
```
Comment (sohaibiftikhar) [inline server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java]: @nik9000 Is this alright or should we do it also for the whole chain?

Comment (nik9000) [reply server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java]: If doing the whole chain makes the PR too big, I'd do this. IT should be fine. If the PR isn't too big then I'd do the whole chain.

Comment (sohaibiftikhar) [reply server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java]: It does make it kind of complicated cause the chain is rather long. For now, I will leave it like this.

**Comment by sohaibiftikhar** [comment]
@nik9000 Could you have a look again when you have time?

**Comment by nik9000** [comment]
> @nik9000 Could you have a look again when you have time?

Sorry! I was distracted with something else. I'll look more closely in the morning. Which is kind of your afternoon. Sorry!

**Review by nik9000 — APPROVED** [review]
This looks right to me. Maybe @talevy should look too because he knows this process better than I do. If he doesn't get a chance to review it by sometime tomorrow I'll give it another look and probably merge it.

File: server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java (hunk 1 above)
Comment (talevy) [reply server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java]: I was going to comment on this, but I understand how much this blows things up. So punting it sounds good!

**Review by talevy — APPROVED** [review]

**Comment by talevy** [comment]
thanks @sohaibiftikhar!

**Pull request merged**